### PR TITLE
W-11932094: Scatter-Gather UnsupportedOperationException with DbAggre…

### PR DESCRIPTION
**Pull request by alopezbizon**
…gate variables

### CODE REVIEW

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java [hunk 1]
```diff
@@ -244,6 +247,46 @@ public void consumablePayload() throws Exception {
     router.process(CoreEvent.builder(testEvent()).message(Message.of(new StringBufferInputStream(TEST_PAYLOAD))).build());
   }
 
+  @Test
+  @Issue("W-11932094")
+  @Description("An unmodifiable list in the first Scatter Gather's route must be able to be merged with the variables of the other routes.")
+  public void mergeVariablesWhenTheFirstRouteHasAnUnmodifiableList() throws Exception {
+    List<String> unmodifiableList = Collections.singletonList("value1");
+    MessageProcessorChain route1 = newChain(empty(), event -> eventWithVariable(event, unmodifiableList));
+    MessageProcessorChain route2 = newChain(empty(), event -> eventWithVariable(event, "value2"));
```
Comment (IvanAndresFritzler) [inline modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java:256]: Extract "value2" to a constant.

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java [hunk 2]
```diff
@@ -33,6 +34,7 @@
 import static org.mule.test.allure.AllureConstants.RoutersFeature.ScatterGatherStory.SCATTER_GATHER;
 import static reactor.core.publisher.Flux.from;
 
+import io.qameta.allure.Issue;
```
Comment (IvanAndresFritzler) [inline modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java:37]: This import should be the last one in the list.

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java [hunk 3]
```diff
@@ -244,6 +247,46 @@ public void consumablePayload() throws Exception {
     router.process(CoreEvent.builder(testEvent()).message(Message.of(new StringBufferInputStream(TEST_PAYLOAD))).build());
   }
 
+  @Test
+  @Issue("W-11932094")
+  @Description("An unmodifiable list in the first Scatter Gather's route must be able to be merged with the variables of the other routes.")
+  public void mergeVariablesWhenTheFirstRouteHasAnUnmodifiableList() throws Exception {
+    List<String> unmodifiableList = Collections.singletonList("value1");
```
Comment (IvanAndresFritzler) [inline modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java:254]: Extract "value1" to a constant.

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java [hunk 4]
```diff
@@ -244,6 +247,46 @@ public void consumablePayload() throws Exception {
...
+    MessageProcessorChain route1 = newChain(empty(), event -> eventWithVariable(event, unmodifiableList));
+    MessageProcessorChain route2 = newChain(empty(), event -> eventWithVariable(event, "value2"));
+
+    muleContext.getInjector().inject(router);
+    router.setRoutes(asList(route1, route2));
+    router.setAnnotations(getAppleFlowComponentLocationAnnotations());
+    router.initialise();
+
+    CoreEvent process = router.process(CoreEvent.builder(testEvent()).message(Message.of(TEST_PAYLOAD)).build());
+    List<String> list = (List) process.getVariables().get("key").getValue();
```
Comment (IvanAndresFritzler) [inline modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java:264]: Extract "key" to a constant.

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java [hunk 5]
```diff
@@ -244,6 +247,46 @@ public void consumablePayload() throws Exception {
...
+    assertThat(list, contains("value1", unmodifiableList));
+  }
+
+  private CoreEvent eventWithVariable(CoreEvent event, Object object) {
```
Comment (IvanAndresFritzler) [inline modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java:286]:
```suggestion
  private CoreEvent addVariable(CoreEvent event, String key, Object value) {
```

**Review by IvanAndresFritzler — CHANGES_REQUESTED** [review]

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java (hunk 1 above)
Comment (alopezbizon) [reply modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java]: Done.

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java (hunk 3 above)
Comment (alopezbizon) [reply modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java]: Done.

File: modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java (hunk 5 above)
Comment (alopezbizon) [reply modules/core-components/src/test/java/org/mule/runtime/core/internal/routing/ScatterGatherRouterTestCase.java]: Done.

**Review by IvanAndresFritzler — APPROVED** [review]

**Pull request merged**